microsoft/PowerToys
# Check spelling 0 0 21

## Summary of the Pull Request

Upgrades check-spelling to [v0.0.21](https://github.com/check-spelling/check-spelling/releases/tag/v0.0.21)

The command to apply changes should now work on Windows (it requires Perl, but I believe that's more or less present most of the time, and it should walk you through the rest of the required tools).


## PR Checklist



## Detailed Description of the Pull Request / Additional comments

There are a bunch of new features, the most important here are probably being able to update the metadata from Windows. (If it doesn't work, please @ me). 

Also, `candidate.patterns` will automatically suggest patterns. You can see them in patterns.txt, e.g.:
```
# Automatically suggested patterns
# hit-count: 2206 file-count: 617
# IServiceProvider
\bI(?=(?:[A-Z][a-z]{2,})+\b)
```

The metadata bits (the hit count/file count) don't have to be retained (I hope they'll be useful in deciding whether/or not to add a pattern, i.e. "how applicable is it?"), the comment hinting at what the pattern does is probably worth retaining. 


## Validation Steps Performed

We've been using more or less this version for a while internally (including talk-to-bot, and, I do have a pattern that could be used to let people use that in forks, but, I'm going to skip that for now).

This weekend, I did some cleanup for `act` (to run check-spelling locally), and some minor polish.

You can see the runs I made in https://github.com/check-spelling/PowerToys/actions -- note that my commit messages had copy-paste errors, so don't rely on them.

## Extra cleanup

Please delete these stale branches:

`dev/aaron-junker/#12637`
`issue_11705_with_example`
`users/niels9001/systrayflyout`
